### DISCUSSION TITLE
Any way to combine controls as the same DPI scaling target?

### DISCUSSION THREAD

**Discussion by styris-ame** [AvaloniaUI/Avalonia#15962] [Q&A] [answered]
I have the following XAML:
```XAML
<Grid Background="Black">
    <Arc x:Name="Arc" StartAngle="-90" SweepAngle="0" Stroke="White" StrokeThickness="5" />

    <Ellipse x:Name="RightEllipse" RenderTransformOrigin="2.5,17.5" Width="5" Height="5" Fill="White" VerticalAlignment="Top"/>
    <Ellipse x:Name="LeftEllipse" RenderTransformOrigin="2.5,17.5" Width="5" Height="5" Fill="White" VerticalAlignment="Top"/>
</Grid>
```
I am animating all three shapes to act as one Arc with rounded edges like so:
![Screenshot 2024-06-08 202623](https://github.com/AvaloniaUI/Avalonia/assets/123859041/e10c4832-743d-4e3b-8bc0-5629d431fce1)

However with low scaling (80%), the elements don't align properly:
![Screenshot 2024-06-08 201705](https://github.com/AvaloniaUI/Avalonia/assets/123859041/c72bd554-e375-4c83-8345-71e21cae698f)

I'm wondering if there is a way to cause the renderer to handle all three shapes as one, while still being able to control the animations of each. I'm open to other ways of tacking this issue too, though making my own Arc class with pre-rounded ends appears very difficult.

**Comment by maxkatz6** [comment, marked as answer]
Check if you have set UseLayoutRounding=false, you will likely need that.

**Reply by styris-ame** [reply]
That fixed it, thank you! I thought it was set to false by default like in WPF, so I didn't think to try that haha

**Comment by thevortexcloud** [comment]
I assume this is for some sort of loading indicator? In which case: Can't you just do this with a single shape and a rounded cap and then just rotate the shape? I have not tried it myself but it looks like all shapes should support rounded caps.

https://github.com/AvaloniaUI/Avalonia/blob/82e3613939af5451597d3c19f3a48e0a8ba888f3/src/Avalonia.Controls/Shapes/Shape.cs#L177-L184

https://github.com/AvaloniaUI/Avalonia/blob/82e3613939af5451597d3c19f3a48e0a8ba888f3/src/Avalonia.Base/Media/PenLineCap.cs#L6

**Reply by styris-ame** [reply]
Thanks for the suggestion. My animation is basically a replica of the new Windows 11 loading animation, where the arc not only rotates, but also shrinks and expands. Using an Arc makes this very easy by simply animating the StartAngle and SweepAngle properties. Perhaps something similar could be done with your suggestion, but it would make things much more complex.
